public/app/core/components/UserAccount/UserAccount.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import Page from 'app/core/components/Page/Page';
+import { StoreState } from 'app/types';
+import UserProfile from './UserProfile';
+import { setIsAdmin } from './state/actions';
+
+export interface Props {
+  admin: string;
+  setIsAdmin: typeof setIsAdmin;
+}
+
+export class UserAccount extends PureComponent<Props> {
+  constructor(props) {
+    super(props);
+
+    this.props.setIsAdmin(props.admin === 'true');
```

Comment:
calling a redux action in constructor is not something you want to do :) if you need to pass state to children use props instead not via redux actions. (have admin prop on the UserProfile component)

Reply:
Also the admin flag is about which api to use right? so we can use this in both Profile & Admin page? Think this flag is not needed on the redux state, but can be only a flag on the component and passed to the load & update profile actions.